src/parser/monk/brewmaster/modules/spells/TigerPalm.tsx

```diff
@@ -0,0 +1,192 @@
+  SPELLS.KEG_SMASH,
+  SPELLS.BLACKOUT_KICK_BRM,
+  SPELLS.BREATH_OF_FIRE,
+  {
+    spell: SPELLS.RUSHING_JADE_WIND,
+    when: (combatant: Combatant) => combatant.hasTalent(SPELLS.RUSHING_JADE_WIND.id) && !combatant.hasBuff(SPELLS.RUSHING_JADE_WIND.id),
+  },
+  {
+    spell: SPELLS.CHI_BURST_TALENT,
+    when: (combatant: Combatant) => combatant.hasTalent(SPELLS.CHI_BURST_TALENT.id),
+  },
+  {
+    spell: SPELLS.CHI_WAVE_TALENT,
+    when: (combatant: Combatant) => combatant.hasTalent(SPELLS.CHI_WAVE_TALENT.id),
+  },
+];
+
+class TigerPalm extends Analyzer {
+  static dependencies = {
+    boc: BlackoutCombo,
+    brews: SharedBrews,
+    statTracker: StatTracker,
+    spellUsable: SpellUsable,
+  };
+
+  protected boc!: BlackoutCombo;
+  protected brews!: SharedBrews;
+  protected statTracker!: StatTracker;
+  protected spellUsable!: SpellUsable;
+
+  totalCasts = 0;
+  badCasts = 0;
+  normalHits = 0;
+  bocHits = 0;
+  lastAttackPower = 0;
+
+  cdr = 0;
+  wastedCDR = 0;
+
+  bocBuffActive = false;
+  bocApplyToTP = false;
+
+  get totalBocHits() {
+    return this.bocHits;
+  }
+
+  get bocEmpoweredThreshold() {
+    if(!this.selectedCombatant.hasTalent(SPELLS.BLACKOUT_COMBO_TALENT.id)) {
+      return null;
+    }
+    return {
+      actual: this.totalBocHits / this.totalCasts,
+      isLessThan: {
+        minor: 0.95,
+        average: 0.9,
+        major: 0.85,
+      },
+      style: ThresholdStyle.PERCENTAGE,
+    };
+  }
+
+  constructor(options: Options){
+    super(options);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.BLACKOUT_COMBO_BUFF), this.onGainBOC);
+    this.addEventListener(Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.BLACKOUT_COMBO_BUFF), this.onGainBOC);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.BLACKOUT_COMBO_BUFF), this.onLoseBOC);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.TIGER_PALM), this.onCast);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.TIGER_PALM), this.checkBadTP);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.TIGER_PALM), this.onDamage);
+    this.addEventListener(Events.damage.to(SELECTED_PLAYER), this.onDamageTaken);
+  }
+
+  onGainBOC(event: ApplyBuffEvent | RefreshBuffEvent){
+    this.bocBuffActive = true;
+  }
+
+  onLoseBOC(event: RemoveBuffEvent){
+    this.bocBuffActive = false;
+  }
+
+  onCast(event: CastEvent) {
+    this.totalCasts += 1;
+    this.bocApplyToTP = this.bocBuffActive;
+  }
+
+  onDamage(event: DamageEvent) {
+    // OK SO we have a hit, lets reduce the CD by the base amount...
+    const actualReduction = this.brews.reduceCooldown(TIGER_PALM_REDUCTION);
+    this.cdr += actualReduction;
+    this.wastedCDR += TIGER_PALM_REDUCTION - actualReduction;
+
+    if (this.bocApplyToTP) {
+      this.bocHits += 1;
+    } else {
+      this.normalHits += 1;
+    }
+  }
+
+  onDamageTaken(event: DamageEvent) {
```

Comment:
What is this doing?

Reply:
Grabbing stack power. It was for face palm relics, where I used ap to determine if it was a FP proc or not, and I guess I missed removing the last vestige.